pontoon/sync/checkouts.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from os import walk
+from os.path import join, normpath, relpath
+from typing import NamedTuple, cast
+
+from django.db.models.manager import BaseManager
+
+from pontoon.base.models import Project, Repository
+from pontoon.sync.repositories import get_repo
+
+log = logging.getLogger(__name__)
+
+
+class Checkout:
+    repo: Repository
+    is_source: bool
+    url: str
+    path: str
+    prev_commit: str | None
+    commit: str | None
+    changed: list[str]
+    """Relative paths from the checkout base"""
+    removed: list[str]
+    """Relative paths from the checkout base"""
+
+    def __init__(self, slug: str, db_repo: Repository, pull: bool) -> None:
+        self.repo = db_repo
+        self.is_source = db_repo.source_repo
+        self.url = db_repo.url
+        self.path = normpath(db_repo.checkout_path)
+
+        if db_repo.last_synced_revisions is None:
+            self.prev_commit = None
+        else:
+            pc = db_repo.last_synced_revisions.get("single_locale", None)
```

Comment:
Agreed. But let's do it as a follow-up; I would much prefer being able to do this refactor without any db changes, so that it's easier to roll back if necessary.